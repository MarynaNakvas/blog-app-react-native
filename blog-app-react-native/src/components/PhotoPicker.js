import React, { useState } from 'react';
import { View, StyleSheet, Image, Button, Alert } from 'react-native';
import * as ImagePicker from 'expo-image-picker';
import * as Permissions from 'expo-permissions';

async function askFoePermissions() {
    const { status } = await Permissions.askAsync(
        Permissions.CAMERA,
        Permissions.CAMERA_ROLL,
    );
    if (status !== 'granted') {
        Alert.alert('Error', 'You did not allow your camera to be used')
        return false;
    }
    return true;
}

export const PhotoPicker = ({}) => {
    const [image, setImage] = useState(null);

    const takePhoto = async () => {
        const hasPermissions = await askFoePermissions();

        if (hasPermissions) {
            return
        }

        const img = await ImagePicker.launchCameraAsync({
            quality: 0.7,
            allowsEditing: false,
            aspect: [16, 9],
        });
        console.log('img', img);
    };

    return (
        <View style={styles.wrapper}>
            <Button
                title='Take photo'
                onPress={takePhoto}
            />
            {image && (
                <Image
                    style={styles.image}
                    source={{uri: image}}
                />
            )}
        </View>
    )
}

const styles = StyleSheet.create({
    wrapper: {
        marginBottom: 10,
    },
    image: {
        width: '100%',
        height: 200,
        marginTop: 10,
    }
})
