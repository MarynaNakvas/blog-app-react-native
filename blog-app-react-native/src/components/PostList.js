import React from 'react';
import { View, StyleSheet, FlatList, Text } from 'react-native';

import { Post } from './Post';

export const PostList = ({ data, onOpen }) => {
    if (!data.length) {
        return (
            <View style={styles.wrapper}>
                <Text style={styles.noItems}>There are no posts</Text>
            </View>
        )
    }

    return (
    <View style={styles.wrapper}>
        <FlatList
            data={data}
            keyExtractor={(post) => post.text}
            renderItem={({ item }) => <Post post={item} onOpen={onOpen} />}
        />
    </View>
)}

const styles = StyleSheet.create({
    wrapper: {
        padding: 10,
    },
    noItems: {
        fontFamily: 'open-regular',
        textAlign: 'center',
        marginVertical: 10,
        fontSize: 18,
    }
})
