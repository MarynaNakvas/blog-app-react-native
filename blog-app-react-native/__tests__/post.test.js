import React from 'react';
import renderer from 'react-test-renderer';

import { Post } from '../src/components/Post';

test('Post renders correctly', () => {
   const post = {
      id: '-N_oQZZWRNOe1JBrE1sx',
      date: '2023-07-20T17:52:10.992Z',
      text: 'Normal text',
      img: 'file:///data/user/0/host.exp.exponent/cache/ExperienceData/%2540marynanakvas%252Fblog-app-react-native/ImagePicker/3758d3f5-1b83-45ca-b709-b8ef528b04b0.jpeg',
      booked: true, 
   };

   const tree = renderer.create(<Post post={post} onOpen={() => null} />).toJSON();

   expect(tree).toMatchSnapshot();
 });
