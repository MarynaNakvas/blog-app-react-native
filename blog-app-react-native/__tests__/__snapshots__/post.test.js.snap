// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "marginBottom": 15,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        {
          "height": 200,
          "width": "100%",
        }
      }
    >
      <Image
        source={
          {
            "uri": "file:///data/user/0/host.exp.exponent/cache/ExperienceData/%2540marynanakvas%252Fblog-app-react-native/ImagePicker/3758d3f5-1b83-45ca-b709-b8ef528b04b0.jpeg",
          }
        }
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "height": 200,
              "width": "100%",
            },
            undefined,
          ]
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0.5)",
            "paddingVertical": 5,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontFamily": "open-regular",
            }
          }
        >
          7/20/2023
        </Text>
      </View>
    </View>
  </View>
</View>
`;
